src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1145,6 +1139,23 @@ private BoundExpression BindSizeOf(SizeOfExpressionSyntax node, DiagnosticBag di
                 this.GetSpecialType(SpecialType.System_Int32, diagnostics, node), hasErrors);
         }
 
+        /// <returns>true if managed type-related errors were found, otherwise false.</returns>
+        private static bool CheckManagedAddr(TypeSymbol type, SyntaxNode node, DiagnosticBag diagnostics)
+        {
+            if (type.IsManagedType)
+            {
+                diagnostics.Add(ErrorCode.ERR_ManagedAddr, node.Location, type);
+                return true;
+            }
+            else if (type.GetArity() != 0)
```

Comment:
I believe this is wrong. A struct with a field of a generic type at any level of nesting should be considered a managed type in 7.3. Should I just determine and cache this information (HasGenericMembers?) in the same traversal that we use to compute IsManagedType? Would appreciate guidance @jaredpar @agocke

Reply:
At a glance this code looks correct to me. Yes in the C# 7.3 and before compiler this would be an error but now it's a warning. That's what I'm expecting though. Am I missing something here?

---
In reply to: [244896797](https://github.com/dotnet/roslyn/pull/31602#discussion_r244896797) [](ancestors = 244896797)

Reply:
Ping on this comment as I didn't see a resolution.

---
In reply to: [245878293](https://github.com/dotnet/roslyn/pull/31602#discussion_r245878293) [](ancestors = 245878293,244896797)

Reply:
This comment was originally referring to a previous revision. Resolved offline.

---
In reply to: [246592720](https://github.com/dotnet/roslyn/pull/31602#discussion_r246592720) [](ancestors = 246592720,245878293,244896797)